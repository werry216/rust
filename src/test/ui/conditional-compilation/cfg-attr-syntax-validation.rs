#[cfg] //~ ERROR `cfg` is not followed by parentheses
struct S1;

#[cfg = 10] //~ ERROR `cfg` is not followed by parentheses
struct S2;

#[cfg()] //~ ERROR `cfg` predicate is not specified
struct S3;

#[cfg(a, b)] //~ ERROR multiple `cfg` predicates are specified
struct S4;

#[cfg("str")] //~ ERROR `cfg` predicate key cannot be a literal
struct S5;

#[cfg(a::b)] //~ ERROR `cfg` predicate key must be an identifier
struct S6;

#[cfg(a())] //~ ERROR invalid predicate `a`
struct S7;

#[cfg(a = 10)] //~ ERROR unsupported literal
struct S8;

#[deprecated(since = b"1.30", note = "hi")] //~ ERROR E0565
struct S9;

macro_rules! generate_s10 {
    ($expr: expr) => {
        #[cfg(feature = $expr)] //~ ERROR `cfg` is not a well-formed meta-item
        struct S10;
    }
}

generate_s10!(concat!("nonexistent"));
