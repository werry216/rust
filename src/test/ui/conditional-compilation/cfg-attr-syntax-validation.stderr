error: `cfg` is not followed by parentheses
  --> $DIR/cfg-attr-syntax-validation.rs:1:1
   |
LL | #[cfg] //~ ERROR `cfg` is not followed by parentheses
   | ^^^^^^ help: expected syntax is: `cfg(/* predicate */)`

error: `cfg` is not followed by parentheses
  --> $DIR/cfg-attr-syntax-validation.rs:4:1
   |
LL | #[cfg = 10] //~ ERROR `cfg` is not followed by parentheses
   | ^^^^^^^^^^^ help: expected syntax is: `cfg(/* predicate */)`

error: `cfg` predicate is not specified
  --> $DIR/cfg-attr-syntax-validation.rs:7:1
   |
LL | #[cfg()] //~ ERROR `cfg` predicate is not specified
   | ^^^^^^^^

error: multiple `cfg` predicates are specified
  --> $DIR/cfg-attr-syntax-validation.rs:10:10
   |
LL | #[cfg(a, b)] //~ ERROR multiple `cfg` predicates are specified
   |          ^

error: `cfg` predicate key cannot be a literal
  --> $DIR/cfg-attr-syntax-validation.rs:13:7
   |
LL | #[cfg("str")] //~ ERROR `cfg` predicate key cannot be a literal
   |       ^^^^^

error: `cfg` predicate key must be an identifier
  --> $DIR/cfg-attr-syntax-validation.rs:16:7
   |
LL | #[cfg(a::b)] //~ ERROR `cfg` predicate key must be an identifier
   |       ^^^^

error[E0537]: invalid predicate `a`
  --> $DIR/cfg-attr-syntax-validation.rs:19:7
   |
LL | #[cfg(a())] //~ ERROR invalid predicate `a`
   |       ^^^

error[E0565]: unsupported literal
  --> $DIR/cfg-attr-syntax-validation.rs:22:11
   |
LL | #[cfg(a = 10)] //~ ERROR unsupported literal
   |           ^^

error: `cfg` is not a well-formed meta-item
  --> $DIR/cfg-attr-syntax-validation.rs:27:9
   |
LL |         #[cfg(feature = $expr)] //~ ERROR `cfg` is not a well-formed meta-item
   |         ^^^^^^^^^^^^^^^^^^^^^^^ help: expected syntax is: `#[cfg(/* predicate */)]`
...
LL | generate_s9!(concat!("nonexistent"));
   | ------------------------------------- in this macro invocation

error: aborting due to 9 previous errors

Some errors occurred: E0537, E0565.
For more information about an error, try `rustc --explain E0537`.
